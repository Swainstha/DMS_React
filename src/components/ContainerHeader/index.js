import React from 'react';
import {Breadcrumb, BreadcrumbItem} from 'reactstrap';

const getDisplayString = (sub) => { //if sub = harry-potter, docs
  const arr = sub.split("-"); // arr[0]= harry , 2= potter
  if (arr.length > 1) {
    return arr[0].charAt(0).toUpperCase() + arr[0].slice(1) + " " + arr[1].charAt(0).toUpperCase() + arr[1].slice(1)
  //Harry Potter
  } else {
    return sub.charAt(0).toUpperCase() + sub.slice(1)  //Docs
  }

};
const getUrlString = (path, sub, index) => { //hello.com/ht/docs/harry-potter, sub=docs(index=2), sub=
  if (index === 0) {
    return '#/';
  } else {
    return '#/' + path.split(sub)[0] + sub; //#/hello.com/ht/docs 
  }
};

const ContainerHeader = ({title, match}) => {
  const path = match.path.substr(1);
  const subPath = path.split('/');
  return (
    <div className="page-heading d-sm-flex justify-content-sm-between align-items-sm-center">
      <h2 className="title mb-3 mb-sm-0">{title}</h2>

      <Breadcrumb className="mb-0" tag="nav">
        {subPath.map((sub, index) => {
            return <BreadcrumbItem active={subPath.length === index + 1}
                                   tag={subPath.length === index + 1 ? "span" : "a"} key={index}
                                   href={getUrlString(path, sub, index)}
                                   data-test="breadCrumb-component">{getDisplayString(sub)}</BreadcrumbItem>
          }
        )}
      </Breadcrumb>
    </div>
  )
};

const test_data="hello.com/ht/docs/harry-potter"
const subData = test_data.split('/');
export default ContainerHeader;
it('getUrlString test', () => {
  expect(getUrlString(test_data, subData[0],0)).toEqual("#/");
  expect(getUrlString(test_data, subData[1],1)).toEqual("#/hello.com/ht");
  expect(getUrlString(test_data, subData[2],2)).toEqual("#/hello.com/ht/docs");
  expect(getUrlString(test_data, subData[3],3)).toEqual("#/hello.com/ht/docs/harry-potter");

});

it('getDisplayString test', () => {
  expect(getDisplayString(subData[0])).toEqual("Hello.com");
  expect(getDisplayString(subData[1])).toEqual("Ht");
  expect(getDisplayString(subData[2])).toEqual("Docs");
  expect(getDisplayString(subData[3])).toEqual("Harry Potter");

});
