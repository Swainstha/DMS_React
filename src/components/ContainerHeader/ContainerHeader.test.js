import React from 'react';
import ContainerHeader from './index';
import {
    configure,
    shallow,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})


const setup = (props = {}) => {
    return shallow( < ContainerHeader {...props}  />
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}


it("ContainerHeader",()=>{
    const wrapper= setup({title:"address", match:{path:"#hello.com/ht/docs/harry-potter"}});
    expect(wrapper).toMatchSnapshot();
})


const dummy_data ="#example.com/hello-ok/chelsea-club"

it("ContainerHeader props children",()=>{
    const wrapper= setup({title:"address", match:{path:dummy_data}});
    const breadCrumb = findByTestAttr(wrapper, "breadCrumb-component");
    const h2= wrapper.find('h2');
    expect(h2).toHaveLength(1);
    expect(h2.text()).toEqual("address");
    expect(breadCrumb).toHaveLength(3);

})