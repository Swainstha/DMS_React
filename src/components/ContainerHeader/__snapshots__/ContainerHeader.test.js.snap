// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerHeader 1`] = `ShallowWrapper {}`;
